packages/core/src/agent/index.ts

```diff
@@ -1452,7 +1451,7 @@ export class Agent<
         const messageList = new MessageList({
           threadId,
           resourceId,
-          generateMessageId,
+          generateMessageId: this.#mastra?.generateId.bind(this.#mastra),
```

Comment:
```suggestion
          generateMessageId: this.#mastra?.generateId?.bind(this.#mastra),
```

Reply:
Gotcha, added that in. Also addressed the `streaming-memory` test failing in the latest commit. I commented out the logic for the `experimental_generateMessageId` on lines 1996-1999. I don't think that's being used, at least not at the moment. Seems like that's possibly something for a future feature where ID generators can be attached to specific primitives? The latest commit also ensures custom IDs are being used for assistant messages as opposed to IDs generated by the AI SDK